apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.UnitTests/ClientPoolTests.cs

```diff
@@ -20,86 +32,85 @@ public ClientPoolTests(ITestOutputHelper outputHelper)
+            var mockClientFactory = SetupMockClientFactory();
 
-            var testPool = new ClientPool(mockClientFactory.Object);
+            var testPool = new ClientPool(mockClientFactory);
             var clientAcquired = await testPool.AcquireClientAsync();
             testPool.ReleaseClient(clientAcquired);
 
-            mockClientFactory.Verify(x => x.CreateClientAsync(It.IsAny<ServiceEndpoint>(), It.IsAny<ITokenAccess>()), Times.Exactly(1));
+            Assert.Equal(1, mockClientFactory.Invocations);
             var s = new StringBuilder();
             Assert.Equal(0, testPool.GetPoolInfo(s));
             Logger.Instance.Info(s.ToString());
         }
 
         [Fact]
-        public async Task SequentialCreatesOnlyUseOneChannel()
+        public async Task ConcurrencyStress()
         {
-            //SpannerClients if created sequentially should only use a single channel.
-            //This is to faciliate good session pool hits (which is per channel).  If we always
-            //round robin'd clients, then we would get hard cache misses until we populated all
-            //of the caches per channel.
-            //Since channels are lazily created, this also saves resources for scenarios like rich clients
-            //that may only need a single session for multiple sequential workloads.
-            //Our default setting for # channels = 4 needs to work well for all scenarios.
-            var firstReturnedClient = new Mock<SpannerClient>();
-            var mockClientFactory = SetupMockClientFactory(firstReturnedClient);
+            const int multiplier = 10;
+            var mockClientFactory = SetupMockClientFactory();
 
-            var testPool = new ClientPool(mockClientFactory.Object);
-            var expectedClient = firstReturnedClient.Object;
+            //A mini stress test that hits the client pool with multiple concurrent client requests.
+            var concurrentQueries = new List<Task<SpannerClient>>();
+            var testPool = new ClientPool(mockClientFactory);
 
-            Assert.Same(expectedClient, await testPool.AcquireClientAsync());
-            testPool.ReleaseClient(expectedClient);
+            for (var i = 0; i < SpannerOptions.Instance.MaximumGrpcChannels * multiplier; i++)
+            {
+                concurrentQueries.Add(GetSpannerClientAsync(testPool));
+            }
 
-            Assert.Equal(expectedClient, await testPool.AcquireClientAsync());
-            testPool.ReleaseClient(expectedClient);
+            await Task.WhenAll(concurrentQueries);
+            Assert.Equal(SpannerOptions.Instance.MaximumGrpcChannels, mockClientFactory.Invocations);
 
-            Assert.Equal(expectedClient, await testPool.AcquireClientAsync());
-            testPool.ReleaseClient(expectedClient);
+            var grouping = concurrentQueries.GroupBy(x => x.Result).ToList();
+            Assert.Equal(SpannerOptions.Instance.MaximumGrpcChannels, grouping.Count());
```

Comment:
Since you create a list, you can use `Count` instead of `Count()`, although it shouldn't matter too much either way, just an extra casting internally.